sys/kern/ringbuf.c

```diff
@@ -107,7 +92,7 @@ int ringbuf_read(ringbuf_t *buf, uio_t *uio) {
     int res = uiomove((char *)buf->data + buf->tail, size, uio);
     if (res)
       return res;
-    consume(buf, size);
+    ringbuf_consume(buf, size);
```

Comment:
@xThaid: that previously mentioned assertion  was confusing to me when I was wondering this usage of `consume`

Reply:
Yeah, the assumption was incorrect.